managing-os/templates/fedora/upgrade-25-to-26.md

```diff
@@ -147,15 +173,125 @@ should be entered (`@dom0` or `@fedora-26`).
+        [user@dom0 ~]$ qvm-run -a fedora-26 gnome-terminal
+
+ 3. Attempt the upgrade process in the new template.
+
+        [user@fedora-26 ~]$ sudo dnf clean all
+        [user@fedora-26 ~]$ sudo dnf --releasever=26 distro-sync --best --allowerasing
+
+    **Note:** `dnf` might ask you to approve importing a new package signing
+    key. For example, you might see a prompt like this one:
+
+        warning: /var/cache/dnf/fedora-d02ca361e1b58501/packages/python2-babel-2.3.4-1.fc26.noarch.rpm: Header V3 RSA/SHA266 Signature, key ID 64dab85d: NOKEY
+        Importing GPG key 0x64DAB85D:
+         Userid     : "Fedora (26) <fedora-26-primary@fedoraproject.org>"
+         Fingerprint: E641 850B 77DF 4353 78D1 D7E2 812A 6B4B 64DA B85D
+         From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-fedora-26-x86_64
+        Is this ok [y/N]:
+
+    This key was already checked when it was installed (notice that the "From"
+    line refers to a location on your local disk), so you can safely say yes to
+    this prompt.
+
+    **Note:** If you encounter no errors, proceed to step 4. If you do encounter
+    errors, see the next two points first.
+
+     * If `dnf` reports that you do not have enough free disk space to proceed
+       with the upgrade process, create an empty file in dom0 to use as a cache
+       and attach it to the template as a virtual disk.
+
+           [user@dom0 ~]$ truncate -s 5GB /var/tmp/template-upgrade-cache.img
+           [user@dom0 ~]$ dev=$(sudo losetup -f --show /var/tmp/template-upgrade-cache.img)
+           [user@dom0 ~]$ qvm-block attach fedora-26 dom0:${dev##*/}
+
+       Then reattempt the upgrade process, but this time use the virtual disk
+       as a cache.
+
+           [user@fedora-26 ~]$ sudo mkfs.ext4 /dev/xvdi
+           [user@fedora-26 ~]$ sudo mount /dev/xvdi /mnt/removable
+           [user@fedora-26 ~]$ sudo dnf clean all
+           [user@fedora-26 ~]$ sudo dnf --releasever=26 --setopt=cachedir=/mnt/removable --best --allowerasing distro-sync
+
+       If this attempt is successful, proceed to step 4.
+
+     * `dnf` may complain:
+
+           At least X MB more space needed on the / filesystem.
+
+       In this case, one option is to [resize the TemplateVM's disk
+       image][resize-disk-image] before reattempting the upgrade process.
+       (See [Additional Information] below for other options.)
+
+ 4. Trim the new template.
+
+        [user@fedora-26 ~]$ sudo fstrim -v /
+
+ 5. Shut down the new TemplateVM (from the command-line or Qubes VM Manager).
+
+        [user@dom0 ~]$ qvm-shutdown fedora-26
+
+ 6. Remove the cache file, if you created one.
+
+        [user@dom0 ~]$ rm /var/tmp/template-upgrade-cache.img
+
+ 7. (Recommended) Switch everything that was set to the old template to the new
+    template, e.g.:
+
+     1. Make the new template the default template:
+
+        R4.0: Menu --> System Tools --> Qubes Global Settings --> Default template
```

Comment:
Lets be consistent: `Qubes 4.0` instead of `R4.0` here. But since this section apply to Qubes 4.0 only, probably the prefix is not needed at all.

Reply:
In that case, I would go with no prefix. But it would be nice to have a name for that menu other than just `Menu`. That doesn't uniquely identify it, since there are a lot of menus.

Reply:
When I hover over "Q" icon, there is "Applications" tooltip. Is it good enough for name?

Reply:
Yes, `Applications Menu` sounds like the right name.